tests/infer/test_autoguide.py

```diff
@@ -250,11 +252,11 @@ def model():
         traced_guide = torch.jit.trace_module(guide, {"call": ()}, check_trace=False)
     torch.jit.save(traced_guide, "/tmp/test_guide_serialization.pt")
     guide_deser = torch.jit.load("/tmp/test_guide_serialization.pt")
-    expected_names = {name for name, _ in guide.named_parameters()}
-    actual_names = {name for name, _ in guide_deser.named_parameters()}
-    assert actual_names == expected_names
-    for name in actual_names:
-        assert_equal(getattr(guide_deser, name), getattr(guide, name).data)
+
+    actual = guide_deser.call()
+    assert len(actual) == len(expected)
+    for name, a, e in zip(names, actual, expected):
+        assert_equal(a, e, msg="{}: {} vs {}".format(name, a, e))
```

Comment:
@neerajprad any ideas why this is failing?

Reply:
Just taking a look.

Reply:
Earlier, it seems like the parameters in `_parameters` and those stored as attributes were both unconstrained. With this PR, the parameters in `_parameters` are still unconstrained but those stored as attributes are constrained (which is probably the expected change?). When this autoguide is serialized and loaded back, both the attributes and the parameters in `_parameters` are unconstrained. In this test, we end up comparing the constrained original parameters with the unconstrained loaded ones. I think you have subsequently modified the test though for debugging.

Reply:
Thanks! I've fixed by 1. resurrecting the `_unconstrained` suffix, and 2. fixing rng seeds before calling `guide.call()` to ensure the randomized guides produce identical results.